pkg/activator/handler/handler.go

```diff
@@ -61,6 +63,38 @@ type activationHandler struct {
 	revisionLister servinglisters.RevisionLister
 	serviceLister  corev1listers.ServiceLister
 	sksLister      netlisters.ServerlessServiceLister
+
+	cache *probeCache
+}
+
+const probeCacheTimout = -5 * time.Second
+
+type probeCache struct {
+	mu     sync.RWMutex
+	clock  clock.Clock
+	probes map[string]time.Time
+}
+
+func newProbeCache() *probeCache {
+	return &probeCache{
+		probes: map[string]time.Time{},
+		clock:  &clock.RealClock{},
+	}
+}
+
+// should returns true if we should probe the URL.
+func (pc *probeCache) should(url string) bool {
+	pc.mu.RLock()
+	defer pc.mu.RUnlock()
+	v, ok := pc.probes[url]
+	return !ok || !v.After(pc.clock.Now().Add(probeCacheTimout))
```

Comment:
I think this can be simplified with `Since` to:

```golang
pc.clock.Since(v) > probeCacheTimeout
```

You can also make probeCacheTimeout positive in that implementation, which is more intuitive I think.

Reply:
Done